From c1e7562809a1ea9955d43bc21fa2b2ac88d95ee5 Mon Sep 17 00:00:00 2001
From: Martin Milata <mmilata@redhat.com>
Date: Fri, 20 Feb 2015 15:50:32 +0100
Subject: [PATCH] tests: fix failure on gcc5 on x86_64

Resolves #1334604

Signed-off-by: Martin Milata <mmilata@redhat.com>
---
 tests/ruby_stacktrace.at | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/tests/ruby_stacktrace.at b/tests/ruby_stacktrace.at
index 5a2d901..09d5638 100644
--- a/tests/ruby_stacktrace.at
+++ b/tests/ruby_stacktrace.at
@@ -261,6 +261,7 @@ main(void)
 
 AT_TESTFUN([sr_ruby_stacktrace_from_json],
 [[
+#include "stacktrace.h"
 #include "ruby/stacktrace.h"
 #include "ruby/frame.h"
 #include "utils.h"
@@ -268,13 +269,14 @@ AT_TESTFUN([sr_ruby_stacktrace_from_json],
 #include <assert.h>
 #include <stdlib.h>
 #include <stdio.h>
+#include <string.h>
 
 void
 check(char *filename)
 {
   char *error_message = NULL;
-  const char *file_contents = sr_file_to_string(filename, &error_message);
-  const char *input = file_contents;
+  char *file_contents = sr_file_to_string(filename, &error_message);
+  char *input = file_contents;
   struct sr_location location;
   sr_location_init(&location);
 
-- 
1.8.3.1

