From 8e872be02ea4f231347aa47f38b4418175c0e43d Mon Sep 17 00:00:00 2001
From: Martin Milata <mmilata@redhat.com>
Date: Wed, 8 Jan 2014 17:27:47 +0100
Subject: [PATCH] unwind: fix build against elfutils-0.158

Signed-off-by: Martin Milata <mmilata@redhat.com>
---
 lib/core_unwind.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/core_unwind.c b/lib/core_unwind.c
index d0c7aec..7910254 100644
--- a/lib/core_unwind.c
+++ b/lib/core_unwind.c
@@ -206,7 +206,7 @@ open_coredump(const char *elf_file, const char *exe_file, char **error_msg)
     ch->cb.section_address = dwfl_offline_section_address;
     ch->dwfl = dwfl_begin(&ch->cb);
 
-    if (dwfl_core_file_report(ch->dwfl, ch->eh) == -1)
+    if (dwfl_core_file_report(ch->dwfl, ch->eh, exe_file) == -1)
     {
         set_error_dwfl("dwfl_core_file_report");
         goto fail_dwfl;
-- 
1.8.3.1

