From 933bd99829000bcd9bf674f0e2e61c562767a09b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marek=20Bry=C5=A1a?= <mbrysa@redhat.com>
Date: Wed, 27 May 2015 14:37:15 +0200
Subject: [PATCH] normalization: add glibc __libc_fatal

---
 lib/normalize.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/lib/normalize.c b/lib/normalize.c
index 4b22645..bec2ccc 100644
--- a/lib/normalize.c
+++ b/lib/normalize.c
@@ -216,7 +216,8 @@ is_removable_glibc_with_above(const char *function_name,
         call_match(function_name, source_file, "__libc_memalign", "malloc.c", NULL) ||
         call_match(function_name, source_file, "__libc_realloc", "malloc.c", NULL) ||
         call_match(function_name, source_file, "__posix_memalign", "malloc.c", NULL) ||
-        call_match(function_name, source_file, "__libc_calloc", "malloc.c", NULL);
+        call_match(function_name, source_file, "__libc_calloc", "malloc.c", NULL) ||
+        call_match(function_name, source_file, "__libc_fatal", "libc", NULL);
 }
 
 static char *
-- 
1.8.3.1

