From 7351e5b5ad04bd840e75fafb8ab19d4e0c90d717 Mon Sep 17 00:00:00 2001
From: Matej Habrnal <mhabrnal@redhat.com>
Date: Fri, 13 May 2016 14:42:23 +0200
Subject: [PATCH] unwinder refresh config h

---
 configure       | 6 +++---
 lib/config.h.in | 4 ++--
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/configure b/configure
index 2418533..f79cfeb 100755
--- a/configure
+++ b/configure
@@ -13261,10 +13261,10 @@ _ACEOF
 
 fi
 
-ac_fn_c_check_func "$LINENO" "dwfl_frame_state_core" "ac_cv_func_dwfl_frame_state_core"
-if test "x$ac_cv_func_dwfl_frame_state_core" = xyes; then :
+ac_fn_c_check_func "$LINENO" "dwfl_getthreads" "ac_cv_func_dwfl_getthreads"
+if test "x$ac_cv_func_dwfl_getthreads" = xyes; then :
 
-$as_echo "#define HAVE_DWFL_FRAME_STATE_CORE /**/" >>confdefs.h
+$as_echo "#define HAVE_DWFL_NEXT_THREAD /**/" >>confdefs.h
 
 fi
 
diff --git a/lib/config.h.in b/lib/config.h.in
index b5c34c6..b7b6255 100644
--- a/lib/config.h.in
+++ b/lib/config.h.in
@@ -6,8 +6,8 @@
 /* Define to 1 if you have the <dwarf.h> header file. */
 #undef HAVE_DWARF_H
 
-/* Have function dwfl_frame_state_core for coredump unwinding */
-#undef HAVE_DWFL_FRAME_STATE_CORE
+/* Have function dwfl_getthreads for coredump unwinding */
+#undef HAVE_DWFL_NEXT_THREAD
 
 /* Define to 1 if you have the <elfutils/libdwfl.h> header file. */
 #undef HAVE_ELFUTILS_LIBDWFL_H
-- 
1.8.3.1

