From f3f36ff858f50aed5a6c4df5144ae15aeb5255b9 Mon Sep 17 00:00:00 2001
From: Martin Milata <mmilata@redhat.com>
Date: Mon, 24 Feb 2014 12:44:12 +0100
Subject: [PATCH] normalize: extend xorg blacklist

Closes rhbz#1068767.

Signed-off-by: Martin Milata <mmilata@redhat.com>
---
 lib/normalize.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/lib/normalize.c b/lib/normalize.c
index d332882..81be188 100644
--- a/lib/normalize.c
+++ b/lib/normalize.c
@@ -164,7 +164,13 @@ is_removable_xorg(const char *function_name,
         call_match(function_name, source_file, "_XReply", "xcb_io.c", NULL) ||
         call_match(function_name, source_file, "_XError", "XlibInt.c", NULL) ||
         call_match(function_name, source_file, "XSync", "Sync.c", NULL) ||
-        call_match(function_name, source_file, "process_responses", "xcb_io.c", NULL);
+        call_match(function_name, source_file, "process_responses", "xcb_io.c", NULL) ||
+        call_match(function_name, source_file, "OsSigHandler", "osinit.c", NULL) ||
+        call_match(function_name, source_file, "FatalError", "log.c", NULL) ||
+        call_match(function_name, source_file, "AbortServer", "log.c", NULL) ||
+        call_match(function_name, source_file, "AbortDDX", "xf86Init.c", NULL) ||
+        call_match(function_name, source_file, "ddxGiveUp", "xf86Init.c", NULL) ||
+        call_match(function_name, source_file, "OsAbort", "utils.c", NULL);
 }
 
 static bool
-- 
1.8.3.1

